### ISSUE TITLE
Merge multiple Note sections into one

### ISSUE THREAD

**Issue by anjali411** [pytorch/pytorch#62339]
## 📚 Documentation

Having separate blocks labeled as "Note" looks pretty noisy and it might be nice to instead collapse all the notes as bullet points under one Note section. Same applies for warnings too!

[
<img width="937" alt="Screen Shot 2021-07-28 at 1 56 42 PM" src="https://user-images.githubusercontent.com/20081078/127372515-9488bb9c-cdd1-41c1-bc71-7545176b77b5.png">
](url)

cc @brianjo @mruberry @jianyuh @nikitaved @pearu @heitorschueroff @walterddr @IvanYashchuk @xwang233 @Lezcano

**Comment by Lezcano** [comment]
Functions in the `torch` namespace are not a particularly good example, as their docs are not very well maintained. Those in the `linalg.` module have some more curated docs.

In the docs of the `linalg` most functions do not have more than one note. The notable exception to this are factorisations, in which we added a number of notes to detail mathematical properties (when does a map exist / is unique / is differentiable).

Now, something that could be interesting would be to indeed merge consecutive notes / warnings and start each of the notes with a short title in bold explaining what each of them is about?

cc @mattip as we discussed this just yesterday.
